File: api-clients/CsharpDotNet2/src/main/CsharpDotNet2/IO/Swagger/Api/StudiesApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using IO.Swagger.Client;
using IO.Swagger.Model;

namespace IO.Swagger.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IStudiesApi
    {
        /// <summary>
        /// Get a list of all current studies. Returns a paginated list of studies.
        /// </summary>
        /// <param name="page">Page to be returned.</param>
        /// <param name="count">The number of items to return. Default 10. Max 50.</param>
        /// <param name="sort">How to sort the items.</param>
        /// <returns>InlineResponse20012</returns>
        InlineResponse20012 GetStudies (int? page, int? count, string sort);
        /// <summary>
        /// Get a list of all current studies for a given condition. Returns a paginated list of studies.
        /// </summary>
        /// <param name="conditionSlug">Slug of the condition to return studies for.</param>
        /// <param name="page">Page to be returned.</param>
        /// <param name="count">The number of items to return. Default 10. Max 50.</param>
        /// <param name="sort">How to sort the items.</param>
        /// <returns>InlineResponse20013</returns>
        InlineResponse20013 GetStudiesByCondition (string conditionSlug, int? page, int? count, string sort);
        /// <summary>
        /// Get a list of all current conditions for studies. Returns a list of all current conditions for studies.
        /// </summary>
        /// <param name="sort">How to sort the items.</param>
        /// <returns>Object</returns>
        Object GetStudiesConditions (string sort);
        /// <summary>
        /// Find study by DOI, PubMed ID, or slug. Returns a single study.
        /// </summary>
        /// <param name="identifierType">Type of identifier to for the study to return.</param>
        /// <param name="identifier">Identifier for the study to return.</param>
        /// <returns>Study</returns>
        Study GetStudyByIdentifier (string identifierType, string identifier);
    }
  
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class StudiesApi : IStudiesApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="StudiesApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public StudiesApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient; 
            else
                this.ApiClient = apiClient;
        }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="StudiesApi"/> class.
        /// </summary>
        /// <returns></returns>
        public StudiesApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }
    
        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }
    
        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }
    
        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}
    
        /// <summary>
        /// Get a list of all current studies. Returns a paginated list of studies.
        /// </summary>
        /// <param name="page">Page to be returned.</param> 
        /// <param name="count">The number of items to return. Default 10. Max 50.</param> 
        /// <param name="sort">How to sort the items.</param> 
        /// <returns>InlineResponse20012</returns>            
        public InlineResponse20012 GetStudies (int? page, int? count, string sort)
        {
            
    
            var path = "/studies";
            path = path.Replace("{format}", "json");
                
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
             if (page != null) queryParams.Add("page", ApiClient.ParameterToString(page)); // query parameter
 if (count != null) queryParams.Add("count", ApiClient.ParameterToString(count)); // query parameter
 if (sort != null) queryParams.Add("sort", ApiClient.ParameterToString(sort)); // query parameter
                                        
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudies: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudies: " + response.ErrorMessage, response.ErrorMessage);
    
            return (InlineResponse20012) ApiClient.Deserialize(response.Content, typeof(InlineResponse20012), response.Headers);
        }
    
        /// <summary>
        /// Get a list of all current studies for a given condition. Returns a paginated list of studies.
        /// </summary>
        /// <param name="conditionSlug">Slug of the condition to return studies for.</param> 
        /// <param name="page">Page to be returned.</param> 
        /// <param name="count">The number of items to return. Default 10. Max 50.</param> 
        /// <param name="sort">How to sort the items.</param> 
        /// <returns>InlineResponse20013</returns>            
        public InlineResponse20013 GetStudiesByCondition (string conditionSlug, int? page, int? count, string sort)
        {
            
            // verify the required parameter 'conditionSlug' is set
            if (conditionSlug == null) throw new ApiException(400, "Missing required parameter 'conditionSlug' when calling GetStudiesByCondition");
            
    
            var path = "/studies/conditions/{conditionSlug}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "conditionSlug" + "}", ApiClient.ParameterToString(conditionSlug));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
             if (page != null) queryParams.Add("page", ApiClient.ParameterToString(page)); // query parameter
 if (count != null) queryParams.Add("count", ApiClient.ParameterToString(count)); // query parameter
 if (sort != null) queryParams.Add("sort", ApiClient.ParameterToString(sort)); // query parameter
                                        
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudiesByCondition: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudiesByCondition: " + response.ErrorMessage, response.ErrorMessage);
    
            return (InlineResponse20013) ApiClient.Deserialize(response.Content, typeof(InlineResponse20013), response.Headers);
        }
    
        /// <summary>
        /// Get a list of all current conditions for studies. Returns a list of all current conditions for studies.
        /// </summary>
        /// <param name="sort">How to sort the items.</param> 
        /// <returns>Object</returns>            
        public Object GetStudiesConditions (string sort)
        {
            
    
            var path = "/studies/conditions";
            path = path.Replace("{format}", "json");
                
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
             if (sort != null) queryParams.Add("sort", ApiClient.ParameterToString(sort)); // query parameter
                                        
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudiesConditions: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudiesConditions: " + response.ErrorMessage, response.ErrorMessage);
    
            return (Object) ApiClient.Deserialize(response.Content, typeof(Object), response.Headers);
        }
    
        /// <summary>
        /// Find study by DOI, PubMed ID, or slug. Returns a single study.
        /// </summary>
        /// <param name="identifierType">Type of identifier to for the study to return.</param> 
        /// <param name="identifier">Identifier for the study to return.</param> 
        /// <returns>Study</returns>            
        public Study GetStudyByIdentifier (string identifierType, string identifier)
        {
            
            // verify the required parameter 'identifierType' is set
            if (identifierType == null) throw new ApiException(400, "Missing required parameter 'identifierType' when calling GetStudyByIdentifier");
            
            // verify the required parameter 'identifier' is set
            if (identifier == null) throw new ApiException(400, "Missing required parameter 'identifier' when calling GetStudyByIdentifier");
            
    
            var path = "/studies/{identifierType}/{identifier}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "identifierType" + "}", ApiClient.ParameterToString(identifierType));
path = path.Replace("{" + "identifier" + "}", ApiClient.ParameterToString(identifier));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudyByIdentifier: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetStudyByIdentifier: " + response.ErrorMessage, response.ErrorMessage);
    
            return (Study) ApiClient.Deserialize(response.Content, typeof(Study), response.Headers);
        }
    
    }
}

File: api-clients/dart/lib/api/studies_api.dart
part of swagger.api;



class StudiesApi {
  final ApiClient apiClient;

  StudiesApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Get a list of all current studies.
  ///
  /// Returns a paginated list of studies.
  Future<InlineResponse20012> getStudies({ int page, int count, String sort }) async {
    Object postBody = null;

    // verify required params are set

    // create path and map variables
    String path = "/studies".replaceAll("{format}","json");

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if(page != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "page", page));
    }
    if(count != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "count", count));
    }
    if(sort != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "sort", sort));
    }
    
    List<String> contentTypes = ["application/json"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'InlineResponse20012') as InlineResponse20012 ;
    } else {
      return null;
    }
  }
  /// Get a list of all current studies for a given condition.
  ///
  /// Returns a paginated list of studies.
  Future<InlineResponse20013> getStudiesByCondition(String conditionSlug, { int page, int count, String sort }) async {
    Object postBody = null;

    // verify required params are set
    if(conditionSlug == null) {
     throw new ApiException(400, "Missing required param: conditionSlug");
    }

    // create path and map variables
    String path = "/studies/conditions/{conditionSlug}".replaceAll("{format}","json").replaceAll("{" + "conditionSlug" + "}", conditionSlug.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if(page != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "page", page));
    }
    if(count != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "count", count));
    }
    if(sort != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "sort", sort));
    }
    
    List<String> contentTypes = ["application/json"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'InlineResponse20013') as InlineResponse20013 ;
    } else {
      return null;
    }
  }
  /// Get a list of all current conditions for studies.
  ///
  /// Returns a list of all current conditions for studies.
  Future<Object> getStudiesConditions({ String sort }) async {
    Object postBody = null;

    // verify required params are set

    // create path and map variables
    String path = "/studies/conditions".replaceAll("{format}","json");

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if(sort != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "sort", sort));
    }
    
    List<String> contentTypes = ["application/json"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'Object') as Object ;
    } else {
      return null;
    }
  }
  /// Find study by DOI, PubMed ID, or slug.
  ///
  /// Returns a single study.
  Future<Study> getStudyByIdentifier(String identifierType, String identifier) async {
    Object postBody = null;

    // verify required params are set
    if(identifierType == null) {
     throw new ApiException(400, "Missing required param: identifierType");
    }
    if(identifier == null) {
     throw new ApiException(400, "Missing required param: identifier");
    }

    // create path and map variables
    String path = "/studies/{identifierType}/{identifier}".replaceAll("{format}","json").replaceAll("{" + "identifierType" + "}", identifierType.toString()).replaceAll("{" + "identifier" + "}", identifier.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = ["application/json"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'Study') as Study ;
    } else {
      return null;
    }
  }
}
